tools/data_integrity/fix_candidate_age.php

```diff
@@ -29,50 +26,72 @@
     $confirm = true;
 }
 
-$incorrectAges = array();
-$nullAges      = array();
+$incorrectAges  = array();
+$nullAges       = array();
+$verifiedTables = array();
 
 foreach ($instruments as $inst=>$fullName) {
 
-    if (!$DB->tableExists($inst)) {
-        echo $inst." does not have a table in the Database\n";
+    //Check what database table is associated with this instrument
+    echo "Instrument: $inst ($fullName)\n";
+    try {
+        $instrument = NDB_BVL_Instrument::factory($inst, "", "");
+    } catch (Exception $e) {
+        echo "\t$inst does not seem to be a valid instrument.\n";
+        continue;
+    }
+
+    $table = $instrument->table;
+
+    if (in_array($table, $verifiedTables, true)) {
+        echo "\tTable $table has already been verified. skipping.\n";
+        continue;
+    } else {
+        //Add table name to repertoire to make sure it is not checked again
+        //in case 2+ instruments use the same table
+        $verifiedTables[] = $table;
+    }
+
+    if (!$DB->tableExists($table)) {
+        echo "\tTable $table for instrument $inst does not exist in the Database\n";
         continue;
     } else if (strpos('_proband', $inst) !== false) {
-        echo $inst." is a Proband instrument and should be handled seperately.\n";
+        echo "\t$inst is a Proband instrument and should be handled seperately.\n";
+        continue;
+    }
+
+    // Break if Date taken does not exist
+    if (!$DB->columnExists($inst, 'Date_taken')) {
+        echo "\t$inst does not use a `Date_taken` field and should be handled separately.\n";
         continue;
     }
 
     //Get instrument SQL table
     $DBInstTable = $DB->pselect(
-        "SELECT * 
-         FROM $inst i 
+        "SELECT i.CommentID, Date_taken, Candidate_Age
+         FROM $table i 
```

Comment:
@ridz1208 For IBIS, we have many survey instruments tables which do have the default examiner, Candidate_Age fields. So the script is failing on IBIS dataset.

Reply:
can you add the fail message here or give me an example table that it fails on ?

Reply:
@ridz1208 Might be good to have a way to check if there is a `Candidate_Age` field in the instrument. Looks like for PREVENT-AD, radiology_review does not have a `Candidate_Age` field making the script to crash with the following error and stopping updating `Candidate_Age` for other instruments:
```
PHP Warning:  PDOStatement::execute(): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'Candidate_Age' in 'field list' in /var/www/loris/php/libraries/Database.class.inc on line 659
Could not execute SELECT i.CommentID, Date_taken, Candidate_Age
         FROM radiology_review i 
            LEFT JOIN flag f ON (i.commentID=f.CommentID)
            LEFT JOIN session s ON (f.SessionID=s.ID)
            LEFT JOIN candidate c ON (s.CandID=c.CandID)
         WHERE c.Active='Y' AND s.Active='Y'. Stack trace#0 /var/www/loris/php/libraries/Database.class.inc(694): Database->execute(Object(PDOStatement), Array)
#1 /var/www/loris/tools/data_integrity/fix_candidate_age.php(77): Database->pselect('SELECT i.Commen...', Array)
#2 {main}
PHP Fatal error:  Uncaught DatabaseException: Unknown column 'Candidate_Age' in 'field list' in /var/www/loris/php/libraries/Database.class.inc:662
Stack trace:
#0 /var/www/loris/php/libraries/Database.class.inc(694): Database->execute(Object(PDOStatement), Array)
#1 /var/www/loris/tools/data_integrity/fix_candidate_age.php(77): Database->pselect('SELECT i.Commen...', Array)
#2 {main}
  thrown in /var/www/loris/php/libraries/Database.class.inc on line 662
```
I would tend to first check that this field is available for the instrument, if not available print a warning, no `Candidate_Age` field for instrument XXX, if available, update the Age

Reply:
@sruthymathew123 @cmadjar thats probably what I'm gonna do

Reply:
Cool! Works now.